docs/diff/rules.md

```diff
@@ -0,0 +1,190 @@
+# Rules for Diff Tool
+
+## Rules to categorize differences
+
+In rules, a difference can be categorized as Breaking or Non-Breaking.
+
+If no rules are provided a default set of rules will be applied on the differences:
+
+|Rule	|Category	|Description	|
+|---	|---	|---	|
+|version-changed	|Breaking	|Categorizes change in the API version	|
+|display-name-changed	|Breaking	|Categorizes change in the display name of any end point operation	|
+|operation-removed	|Breaking	|Categorizes removal of any end point operation	|
+|parameter-removed	|Breaking	|Categorizes removal of any parameter (path, query, etc. ) in any end point operation	|
+|required-parameter-added	|Breaking	|Categorizes addition of any required parameter (path, query, etc. ) in any end point operation	|
+|example-changed	|Ignored	|Catagorizes any changes to the API spec examples	|
+|operation-added	|Non-Breaking	|Categorizes addition of endpoint operation	|
+
+## Defining your own rules
+
+You can define your own rule set and provide it to the diff command. A rule set is defined as a json array. When you provide your own rule set the default rule set will not be applied.
+
+To define rules, you need to understand how differences are generated.
+
+1. Using [AMF](https://www.npmjs.com/package/amf-client-js), a flattened JSON-LD graph is generated for the the given APIs.
+2. Differences are calculated for each node in the JSON-LD based on the node ID.
+3. Then rules are applied on each node.
+
+### **Example**
+
+Below are the differences generated for version change and addition of a required parameter.
```

Comment:
Suggest changing to 'Here's an example of the differences when changing the version or adding a required parameter.'